### PR TITLE
[terraform/modules/ignition] Adds support for arbitary ignition config

### PR DESCRIPTION
**Pull request by captn3m0**
The only 2 real changes are:

1. Default to `"qmachu/etcd-cloud-operator:v3.3.3"` instead of `:latest`
2. Adds a `ignition_extra_config`

Tested :checkered_flag: 

~This is still in testing, but I'll confirm back soon.~

### CODE REVIEW

File: terraform/common.tf [hunk 1]
```diff
@@ -30,12 +30,12 @@
 
 variable "instance_ssh_keys" {
   description = "List of SSH public keys that are allowed to login into nodes"
-  type = "list"
+  type        = "list"
 }
 
 variable "eco_image" {
   description = "Container image of ECO to use"
-  default     = "qmachu/etcd-cloud-operator:latest"
```
Comment (Quentin-M) [inline terraform/common.tf]: Would you mind making that `v3.3.3b`? `v3.3.3` has a nasty bug.

File: terraform/modules/ignition/ignition.tf [hunk 2]
```diff
@@ -32,10 +32,15 @@ data "ignition_config" "main" {
   ]
 
   users = ["${data.ignition_user.core.id}"]
+
+  append {
+    source       = "${lookup(var.ignition_extra_config, "source")}"
+    verification = "${lookup(var.ignition_extra_config, "verification", "")}"
+  }
```
Comment (Quentin-M) [inline terraform/modules/ignition/ignition.tf]: I assume that the `ignition_config` resource still behaves properly if those are empty?

Comment (captn3m0) [reply terraform/modules/ignition/ignition.tf]: [Filed a bug report](https://github.com/terraform-providers/terraform-provider-ignition/issues/48) against the provider, and using a hacky solution for now (use a empty config).

Comment (Quentin-M) [reply terraform/modules/ignition/ignition.tf]: Awesome, thanks! Happy to merge once the other comment is addressed as well.

**Comment by Quentin-M** [comment]
Thank you for that~!

**Pull request merged**